SuperLU-Dist: Add CUDA variant

pghysels opened the pull request:
Add CUDA variant to SuperLU-dist 

Only enabled for latest version 6.4.0.
This version of SuperLU-Dist does not (yet) have any CUDA kernels, but uses the CUDA library cuBLAS.

pghysels committed:
Add CUDA support to superlu-dist

adamjstewart commented on var/spack/repos/builtin/packages/superlu-dist/package.py:
```diff
@@ -78,6 +80,15 @@ def cmake_args(self):
 
+        if '+cuda' in spec:
+            args.append('-DTPL_ENABLE_CUDALIB=TRUE')
+            args.append('-DTPL_CUDA_LIBRARIES=-L%s -lcublas -lcudart'
+                        % spec['cuda'].prefix.lib)
```
```suggestion
            args.append('-DTPL_CUDA_LIBRARIES=' + 
                        spec['cuda'].libs.ld_flags)
```
This will work even if the libs are installed in `lib64` instead of `lib`.

balay replied:
This patch does not work. I get:
```
'-DTPL_CUDA_LIBRARIES=-L/data/balay/spack-xsdk-slu2/opt/spack/linux-centos7-skylake_avx512/gcc-7.4.0/cuda-10.2.89-xdkte4bqy4qbf5wqr2r3wxgen6uuiiuv/lib64 -lcudart'
```
i.e `-lcublas` is missing.

adamjstewart replied:
Should we add `libcublas` to the `libs` method of the CUDA package? @ax3l @Rombur

ax3l replied:
Ouch, probably not a bad idea.

ax3l replied:
I wonder why this is needed, this is a CMake package and CMake is quite good in picking these libs up.

pghysels replied:
SuperLU_dist's CMakeLists.txt does not look for cuBLAS, it expects the user to set `-DTPL_CUDA_LIBRARIES`

adamjstewart reviewed (changes requested):
@xiaoye @gchavez2 @balay can you review?

pghysels committed:
Use spec['cuda'].libs.directories[0] iso spec['cuda'].prefix.lib
so it works for both lib and lib64

The suggested:
args.append('-DTPL_CUDA_LIBRARIES=' +
            spec['cuda'].libs.ld_flags)
did not work because it does not link with cuBLAS.

balay reviewed (approved):
works for me

And the default is ~cuda - so the current xsdk build is unchanged.
